assets/src/blocks/Spreadsheet/SpreadsheetFrontend.js

```diff
@@ -0,0 +1,186 @@
+import { Component, Fragment } from '@wordpress/element';
+import { ArrowIcon } from './ArrowIcon';
+
+const {apiFetch} = wp;
+const {addQueryArgs} = wp.url;
+const {__} = wp.i18n;
+
+export class SpreadsheetFrontend extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      loading: true,
+      sheet_data: null,
+      invalidSheetId: false,
+      searchText: '',
+      sortDirection: 'asc',
+      sortColumnIndex: 0,
+    };
+
+    this.onHeaderClick = this.onHeaderClick.bind(this);
+  }
+
+  onHeaderClick( index ) {
+    if ( index == this.state.sortColumnIndex ) {
+      const newDirection = this.state.sortDirection == 'asc' ? 'desc' : 'asc';
+      this.setState({
+        sortDirection: newDirection
+      });
+    } else {
+      this.setState({
+        sortColumnIndex: index,
+        sortDirection: 'asc',
+      });
+    }
+  }
+
+  componentDidMount() {
+    const sheetID = this.extractSheetID( this.props.url );
+
+    if (sheetID) {
+      const queryArgs = {
+        path: addQueryArgs('/planet4/v1/get-spreadsheet-data', {
+          sheet_id: sheetID,
+        })
+      };
+
+      apiFetch( queryArgs )
+        .then( sheet_data => {
+          this.setState( { loading: false, sheet_data } );
+      } );
+    }
+  }
+
+  extractSheetID( url ) {
+    const googleSheetsPattern = /docs\.google\.com\/spreadsheets\/d\/e\/([\w-]+)/;
+    const matches = url.match(googleSheetsPattern);
+    if (matches !== null) {
+      return matches[1];
+    } else {
+      this.setState( { invalidSheetId: true } );
+    }
+  }
+
+  sortRows( rows, columnIndex ) {
+    const sortedRows = rows.sort( ( rowA, rowB ) => {
+      const textCompare = rowA[columnIndex].localeCompare( rowB[columnIndex] );
+      if ( textCompare !== 0 ) {
+        return textCompare;
+      }
+    } );
+    if (this.state.sortDirection == 'desc') {
+      return sortedRows.reverse();
+    } else {
+      return sortedRows;
+    }
+  };
+
+  filterMatchingRows( rows ) {
+    const filteredRows = rows.filter( row => {
+      return !!row.find( cell => cell.toLowerCase().includes( this.state.searchText.toLowerCase() ) );
+    })
+    return filteredRows;
+  }
+
+  boldText( text ) {
+    return <b>{ text }</b>;
+  }
+
+  highlightMatches( cellValue, searchText ) {
+    let reg = new RegExp('(' + searchText + ')', 'gi')
```

Comment:
Actually that was my first suggestion when we estimated PLANET-4914 on a backlog refinement call, but you suggested doing it ourselves to avoid including a library as it would be pretty simple... which actually made sense, I just did a quick test adding the library and the `frontendIndex` build went from 7.2K to 24K... I think we can keep this version unless we need extra features as the ones provided by that library or we start using highlighting in other places...

Reply:
Right, I did suggest that :) Yeah you're right, that increase in frontend footprint doesn't seem worth it if it's our only use case. Still would be nicer if we extracted the highlighting logic from `SpreadsheetFrontend` so we can keep that one a bit more high level.